### ISSUE TITLE
team tab hide issues

### ISSUE THREAD

**Issue by caver456** [ncssar/radiolog#766]
See previous related issues (all closed) #741, #728, #702, #670, #406.

This issue is new after fixing of #728, since the symptoms are a bit different than they were in 3.12.0.

Actions needed to reproduce:

1) create an entry for 'Team 101'
2) create an entry for '1p42'
  --> notice that extTeamNameList after sort is ['spacerLeft', 'z_01p42', 'spacer1', 'z_Team00101', 'spacer2'] but the tab bar doesn't show a spacer - this might be related as below, but should be a separate issue if it still happens after closing of this issue: 
![image](https://github.com/user-attachments/assets/36ef680f-e2a9-412e-8167-a4b47b64250c)
3) hide tab '1p42'

which causes this - probably due to the back-to-back spacers when there is no spacer visible:

```
063322:deleting team tab: teamName=1p42  extTeamName=z_01p42  niceTeamName=1p42
063322:  teamNameList before deletion:['dummy', '1p42', 'Team 101']
063322:  extTeamNameList before deletion:['spacerLeft', 'z_01p42', 'spacer1', 'z_Team00101', 'spacer2']
063322:  i=1
063322:  deleted proxyModelList index 1
063322:Freeing hotkey '2' which was used for callsign '1p42'
063322:checking tab#1:Team 101
063322:checking tab#2:Spacer 1
063322:  found back-to-back spacers at indices 0 and 1
063322:deleting team tab: teamName=spacer1  extTeamName=spacer1  niceTeamName=Spacer 1
063322:  teamNameList before deletion:['dummy', 'Team 101']
063322:  extTeamNameList before deletion:['spacerLeft', 'spacer1', 'z_Team00101', 'spacer2']
063322:  i=1
063322:  deleted proxyModelList index 1
Traceback (most recent call last):
  File "C:\Users\caver\Documents\GitHub\radiolog\radiolog.py", line 5592, in tabContextMenu
    self.deleteTeamTab(niceTeamName)
  File "C:\Users\caver\Documents\GitHub\radiolog\radiolog.py", line 6102, in deleteTeamTab
    if self.extTeamNameList[n+1].lower().startswith("spacer"):
       ~~~~~~~~~~~~~~~~~~~~^^^^^
IndexError: list index out of range
```

Then, from the sidebar, click and unhide 1p42, which starts these repeating trracebacks every second:

```
063553:newEntry called with these values:
063553:['0635', '', '1p42', '[RADIOLOG: operator is unhiding hidden team tab for "1p42"]', '', 'Working', 1727012153.172847, '', '', '']
063553:new team: newTeamName=1p42 extTeamName=z_01p42 niceTeamName=1p42 shortNiceTeamName=1p42
063553:extTeamNameList before sort:['spacerLeft', 'z_Team00101', 'spacer2', 'z_01p42']
063553:grouped tabs:{'NCSOcmd': [], 'NCSO': ['z_01p42'], 'NCPD': [], 'TPD': [], 'GVPD': [], 'CHP': [], 'Numbers': ['z_Team00101'], 'other': []}
063553:extTeamNameList after sort:['spacerLeft', 'z_01p42', 'spacer1', 'z_Team00101', 'spacer2']
063553:setting tab initial style in addTab: #tab_z_01p42 { font-size:20px;background:none;padding-left:1px;padding-right:1px }
063553:next available hotkey:2
063553:ERROR in updateTeamTimers: attempted to update appearance for a non-existent tab for z_Team00101
Traceback (most recent call last):
  File "C:\Users\caver\Documents\GitHub\radiolog\radiolog.py", line 4063, in updateTeamTimers
    f=self.ui.tabWidget.tabBar().tabButton(i,QTabBar.LeftSide).font()
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'font'
```

**Comment by caver456** [comment]
Adding some more log lines in deleteTeamTab reveals that this looks like a recursive logic goof.

```
		rprint("  extTeamNameList after delete 1: "+str(self.extTeamNameList))
		# if there are two adjacent spacers, delete the second one
		for n in range(len(self.extTeamNameList)-1):
			if self.extTeamNameList[n].lower().startswith("spacer"):
				if self.extTeamNameList[n+1].lower().startswith("spacer"):
					if len(self.extTeamNameList)>2: # but not if there are no visible tabs
						rprint("  found back-to-back spacers at indices "+str(n)+" and "+str(n+1))
						self.deleteTeamTab(self.extTeamNameList[n+1],True)
						rprint('    extTeamNameList on return from recursive deleteTeamTab call:'+str(self.extTeamNameList))
		# 741 call rebuildGroupedTabDict to make sure spacer(s) are in appropriate spots after deleting a team
		rprint("  extTeamNameList after delete 2: "+str(self.extTeamNameList))
```

trying the same script:

```
070150:deleting team tab: teamName=1p42  extTeamName=z_01p42  niceTeamName=1p42
070150:  teamNameList before deletion:['dummy', '1p42', 'Team 101']
070150:  extTeamNameList before deletion:['spacerLeft', 'z_01p42', 'spacer1', 'z_Team00101', 'spacer2']
070150:  i=1
070150:  deleted proxyModelList index 1
070150:Freeing hotkey '2' which was used for callsign '1p42'
070150:checking tab#1:Team 101
070150:checking tab#2:Spacer 1
070150:  extTeamNameList after delete 1: ['spacerLeft', 'spacer1', 'z_Team00101', 'spacer2']
070150:  found back-to-back spacers at indices 0 and 1
070150:deleting team tab: teamName=spacer1  extTeamName=spacer1  niceTeamName=Spacer 1
070150:  teamNameList before deletion:['dummy', 'Team 101']
070150:  extTeamNameList before deletion:['spacerLeft', 'spacer1', 'z_Team00101', 'spacer2']
070150:  i=1
070150:  deleted proxyModelList index 1
070150:  extTeamNameList after delete 1: ['spacerLeft', 'z_Team00101', 'spacer2']
070150:  extTeamNameList after delete 2: ['spacerLeft', 'z_Team00101', 'spacer2']
070150:    extTeamNameList on return from recursive deleteTeamTab call:['spacerLeft', 'z_Team00101', 'spacer2']
Traceback (most recent call last):
  File "C:\Users\caver\Documents\GitHub\radiolog\radiolog.py", line 5592, in tabContextMenu
    self.deleteTeamTab(niceTeamName)
  File "C:\Users\caver\Documents\GitHub\radiolog\radiolog.py", line 6102, in deleteTeamTab
    if self.extTeamNameList[n+1].lower().startswith("spacer"):
       ~~~~~~~~~~~~~~~~~~~~^^^^^
IndexError: list index out of range
```

At the start of the for loop, the length of extTeamNameList is 4.  But, extTeamNameList gets shortened due to the recursive call, so, by the time the culprit line is reached on the next iteration after the recursive call (still inside the for loop which is doing 4 iterations), extTeamNameList only has 3 elements.

**Comment by caver456** [comment]
Adding 'break' after the recursive call gets rid of the traceback, but there are two problems with this approach:
1) it assumes there are no additional occurrences of back-to-back spacers; this >should< be safe...?
2) both team tabs (1p42 and 101) get deleted from the tab bar (!) though the sidebar only shows 1p42 as being deleted; then unhiding 1p42 from the sidebar, as in the initial issue description, still starts the same every-second tracebacks

**Comment by caver456** [comment]
Looking just at the first part of problem 2 (both tabs get hidden though the sidebar only thinks 1p42 is hidden): here's the transcript beginning with the hide of 1p42:

```
072050:tab context menu requested: pos=PyQt5.QtCore.QPoint(60, 13)
072050:  i=1  tabRect=PyQt5.QtCore.QPoint(20, 35):PyQt5.QtCore.QPoint(81, 0)
072050:  extTeamName=z_01p42  niceTeamName=1p42
072051:deleteTeamTabAction clicked
072051:deleting team tab: teamName=1p42  extTeamName=z_01p42  niceTeamName=1p42
072051:  teamNameList before deletion:['dummy', '1p42', 'Team 101']
072051:  extTeamNameList before deletion:['spacerLeft', 'z_01p42', 'spacer1', 'z_Team00101', 'spacer2']
072051:  i=1
072051:  deleted proxyModelList index 1
072051:Freeing hotkey '2' which was used for callsign '1p42'
072051:checking tab#1:Team 101
072051:checking tab#2:Spacer 1
072051:  extTeamNameList after delete 1: ['spacerLeft', 'spacer1', 'z_Team00101', 'spacer2']
072051:  found back-to-back spacers at indices 0 and 1
072051:deleting team tab: teamName=spacer1  extTeamName=spacer1  niceTeamName=Spacer 1
072051:  teamNameList before deletion:['dummy', 'Team 101']
072051:  extTeamNameList before deletion:['spacerLeft', 'spacer1', 'z_Team00101', 'spacer2']
072051:  i=1
072051:  deleted proxyModelList index 1
072051:  extTeamNameList after delete 1: ['spacerLeft', 'z_Team00101', 'spacer2']
072051:  extTeamNameList after delete 2: ['spacerLeft', 'z_Team00101', 'spacer2']
072051:    extTeamNameList on return from recursive deleteTeamTab call:['spacerLeft', 'z_Team00101', 'spacer2']
072051:  extTeamNameList after delete 2: ['spacerLeft', 'z_Team00101', 'spacer2']
```
Nothing seems obviously wrong here... need to keep digging.  What actually modifies the tab bar widget?

**Comment by caver456** [comment]
These lines earlier in deleteTeamTab do the actual deletions.

```
			del self.ui.tabList[i]
			del self.ui.tabGridLayoutList[i]
			del self.ui.tableViewList[i]
			self.ui.tabWidget.removeTab(i)
```

The problem is probably the spacer that doesn't show up, from the initial description: these statements are trying to get rid of the visible spacer, but, it was never actually there, so, they get rid of the thig after it instead (Team 101).

So maybe the easier thing to try is to track down why the spacer doesn't get drawn in the first place.  That won't address the liability that the lists can get out of sync with each other, so a more robust fix would be great in the long-term.

**Comment by caver456** [comment]
addTeam does know that 1p42 is in its own group.  extTeamNameList gets updated accordingly (spacer between groups), but, notice that the spacer names get shuffled: spacer1 was to the right of Team 101, at the end, but now spacer1 is between groups, and spacer2 is at the end.  Maybe spacer1 doesn't get drawn during add of 1p42 because it was already drawn during add of 101?

```
072038:Accepted
072038:newEntry called with these values:
072038:['0720', 'FROM', 'Team 101', 'STARTING ASSIGNMENT', None, 'Working', 1727014837.3607368, None, None, None]
072038:new team: newTeamName=Team 101 extTeamName=z_Team00101 niceTeamName=Team 101 shortNiceTeamName=101
072038:extTeamNameList before sort:['dummy', 'z_Team00101']
072038:grouped tabs:{'NCSOcmd': [], 'NCSO': [], 'NCPD': [], 'TPD': [], 'GVPD': [], 'CHP': [], 'Numbers': ['z_Team00101'], 'other': ['dummy']}
072038:extTeamNameList after sort:['spacerLeft', 'z_Team00101', 'spacer1']
072038:setting tab initial style in addTab: #tab_spacerLeft { font-size:20px;background:none;padding-left:1px;padding-right:1px }
072038:setting tab initial style in addTab: #tab_z_Team00101 { font-size:20px;background:none;padding-left:1px;padding-right:1px }
072038:teamNotesBuildTooltip called for z_Team00101
072038:setting tab initial style in addTab: #tab_spacer1 { font-size:20px;background:none;padding-left:1px;padding-right:1px }
072038:teamNotesBuildTooltip called for spacer1
072038:next available hotkey:1
072038:Accepted2
...
072046:Accepted
072046:newEntry called with these values:
072046:['0720', 'FROM', '1p42', 'STARTING ASSIGNMENT', None, 'Working', 1727014842.416682, None, None, None]
072046:new team: newTeamName=1p42 extTeamName=z_01p42 niceTeamName=1p42 shortNiceTeamName=1p42
072046:extTeamNameList before sort:['spacerLeft', 'z_Team00101', 'spacer1', 'z_01p42']
072046:grouped tabs:{'NCSOcmd': [], 'NCSO': ['z_01p42'], 'NCPD': [], 'TPD': [], 'GVPD': [], 'CHP': [], 'Numbers': ['z_Team00101'], 'other': []}
072046:extTeamNameList after sort:['spacerLeft', 'z_01p42', 'spacer1', 'z_Team00101', 'spacer2']
072046:setting tab initial style in addTab: #tab_z_01p42 { font-size:20px;background:none;padding-left:1px;padding-right:1px }
072046:next available hotkey:2
072046:Accepted2
```

**Comment by caver456** [comment]
Interesting: in a new session, if 1p42 is added first, and Team 101 is added next:
- the spacer between groups appears as expected
- hiding 1p42 works as expected (it does not also hide 101)
- the sidebar shows as expected (hidden is in brackets and italics, unhidden is plain font)
- BUT: after clicking 1p42 in the sidebar, and confirming to unhide it, the same once-a-second tracebacks begin.

**Comment by caver456** [comment]
#670, which primarily dealt with lag reduction, got into the same details as this issue:  In newTeam, the old code (before #670 fix) would always call rebuildTabs on every call to newTeam (when not in the middle of loading a file).  The lag fix was to only call rebuildTabs for certain conditions, and the rest of the time to only call the individual addTab.  This is the root of the problem for this issue - confirmed by changing to call rebuildTabs always, which fixes all the issues.

The current code inside newTeam, with some print lines added:
```
		if not self.loadFlag:
			# 670 - for the first team of a new session, and when unhiding the only tab
			#  (due to display issues noted in https://github.com/ncssar/radiolog/issues/670#issuecomment-1712814526)
			#  follow the old behavior and rebuild the entire tab bar;
			#  for subsequent teams, only add the tab for the new team
			# 741 - also rebuild tabs if restoring a hidden team tab
			rprint('restoringHidden='+str(restoringHidden)+'  prevGroupCount='+str(prevGroupCount)+'  nonEmptyTabGroupCount='+str(self.nonEmptyTabGroupCount)+'  tabBar.count='+str(self.ui.tabWidget.tabBar().count()))
			if restoringHidden or prevGroupCount!=self.nonEmptyTabGroupCount or self.ui.tabWidget.tabBar().count()<2:
				rprint('t1')
				self.rebuildTabs()
			else:
				# display issues when unhiding the rightmost tab
				#  https://github.com/ncssar/radiolog/issues/670#issuecomment-1712814526
				# so, if this would be the rightmost tab, activate a different tab first
				self.ui.tabWidget.tabBar().setCurrentIndex(0)
				rprint('t2: extTeamName='+str(extTeamName))
				self.addTab(extTeamName)
```

The 'if restoringHidden...' logic is meant to call rebuildTabs whenever a group is added or removed, but, the added print lines show that the logic doesn't catch the case where 'dummy' is removed from the 'other' group, resulting in no change in non-empty group count when 1p42 is added:

```
144614:Accepted
144614:newEntry called with these values:
144614:['1446', 'FROM', 'Team 101', 'STARTING ASSIGNMENT', None, 'Working', 1727041570.6200132, None, None, None]
144614:new team: newTeamName=Team 101 extTeamName=z_Team00101 niceTeamName=Team 101 shortNiceTeamName=101
144614:extTeamNameList before sort:['dummy', 'z_Team00101']
144614:grouped tabs:{'NCSOcmd': [], 'NCSO': [], 'NCPD': [], 'TPD': [], 'GVPD': [], 'CHP': [], 'Numbers': ['z_Team00101'], 'other': ['dummy']}
144614:extTeamNameList after sort:['spacerLeft', 'z_Team00101', 'spacer1']
144614:restoringHidden=False  prevGroupCount=0  nonEmptyTabGroupCount=2  tabBar.count=0
144614:t1
...
144618:Accepted
144618:newEntry called with these values:
144618:['1446', 'FROM', '1p42', 'STARTING ASSIGNMENT', None, 'Working', 1727041575.0196114, None, None, None]
144618:new team: newTeamName=1p42 extTeamName=z_01p42 niceTeamName=1p42 shortNiceTeamName=1p42
144618:extTeamNameList before sort:['spacerLeft', 'z_Team00101', 'spacer1', 'z_01p42']
144618:grouped tabs:{'NCSOcmd': [], 'NCSO': ['z_01p42'], 'NCPD': [], 'TPD': [], 'GVPD': [], 'CHP': [], 'Numbers': ['z_Team00101'], 'other': []}
144618:extTeamNameList after sort:['spacerLeft', 'z_01p42', 'spacer1', 'z_Team00101', 'spacer2']
144618:restoringHidden=False  prevGroupCount=2  nonEmptyTabGroupCount=2  tabBar.count=3
144618:t2: extTeamName=z_01p42
```
One possible fix is to keep track of a list of non empty tab group names, rather than just the count of non empty tab groups (for both prevGroupCount and nonEmptyTabGroupCount).  If the lists are not identical, call rebuildTabs instead of individual addTab.

**Comment by caver456** [comment]
Close!  Comparing the actual list of non empty tab groups does partially work: Team 101, then 1p42, does show the spacer after 1p42.  And hiding 1p42 works as expected (only 1p42 is hidden).

But, on unhide of 1p42, no spacer appears after 1p42.  This is probably because nonEmptyTabGroups isn't updated during deleteTeamTab, and hiddenTeamTabsList isn't checked to see if any tabs in a group are actually displayed.

Maybe we need another list, nonEmptyDisplayedTabGroups or such.  Oy.

**Comment by caver456** [comment]
Decided the cleanest and most robust way is to pass an argument 'unhiding' all the way from unhideTeamTab thru to newTeam.  This way, there's no need to calculate restoringHidden inside newTeam, which is a faulty calculation anyway since the extTeamName has already been removed from hiddenTeamTabsList twice by that time (in unhideTeamTab and in addTab).

This seems to work - but some more thorough QA of obscure edge cases would be good.

**Comment by caver456** [comment]
Found a problem during unhide-by-new-typed-callsign-entry for Team 101, after having tried various hide-and-unhide of callsigns from two different tab groups (1p42 and 4p13) - this initial traceback is followed by the once-a-second 'font' tracebacks:

```
172509:Accepted
172509:newEntry called with these values:
172509:['1725', 'FROM', 'Team 101', 'STARTING ASSIGNMENT', None, 'Working', 1727051107.894923, None, None, None]
172509:new team: newTeamName=Team 101 extTeamName=z_Team00101 niceTeamName=Team 101 shortNiceTeamName=101
172509:extTeamNameList before sort:['spacerLeft', 'z_04p13', 'spacer2', 'z_Team00101']
172509:grouped tabs:{'NCSOcmd': [], 'NCSO': ['z_01p42'], 'NCPD': [], 'TPD': [], 'GVPD': ['z_04p13'], 'CHP': [], 'Numbers': ['z_Team00101'], 'other': []}
172509:extTeamNameList after sort:['spacerLeft', 'spacer1', 'z_04p13', 'spacer2', 'z_Team00101', 'spacer3']
Traceback (most recent call last):
  File "C:\Users\caver\Documents\GitHub\radiolog\radiolog.py", line 7698, in keyPressEvent
    self.accept()
  File "C:\Users\caver\Documents\GitHub\radiolog\radiolog.py", line 7851, in accept
    self.parent.newEntry(val,self.amendFlag)
  File "C:\Users\caver\Documents\GitHub\radiolog\radiolog.py", line 5002, in newEntry
    self.newEntryProcessTeam(niceTeamName,status,values[1],values[3],amend,unhiding=unhiding)
  File "C:\Users\caver\Documents\GitHub\radiolog\radiolog.py", line 5017, in newEntryProcessTeam
    self.newTeam(niceTeamName,unhiding=unhiding)
  File "C:\Users\caver\Documents\GitHub\radiolog\radiolog.py", line 5246, in newTeam
    self.addTab(extTeamName)
  File "C:\Users\caver\Documents\GitHub\radiolog\radiolog.py", line 5375, in addTab
    self.ui.tabList[i].setStyleSheet('font-size:'+str(self.menuFontSize)+'pt')
    ~~~~~~~~~~~~~~~^^^
IndexError: list index out of range
```

Confirmed that this does not happen in the trivial case (Team 101 is the only callsign); it does not happen after hiding 1p42; it does not happen after creating 1p42 and 4p13 and then 101, then hiding 101; it does not happen after this one-time error while hiding 101:

```
173534:deleting team tab: teamName=Team 101  extTeamName=z_Team00101  niceTeamName=Team 101
173534:  teamNameList before deletion:['dummy', '1p42', 'Team 101']
173534:  extTeamNameList before deletion:['spacerLeft', 'z_01p42', 'spacer1', 'z_Team00101', 'spacer3']
173534:  i=3
173534:  deleted proxyModelList index 3
173534:Freeing hotkey '3' which was used for callsign 'Team 101'
173534:checking tab#1:1p42
173534:checking tab#2:Spacer 1
173534:  found back-to-back spacers at indices 2 and 3
173534:deleting team tab: teamName=spacer3  extTeamName=spacer3  niceTeamName=Spacer 3
173534:  teamNameList before deletion:['dummy', '1p42']
173534:  extTeamNameList before deletion:['spacerLeft', 'z_01p42', 'spacer1', 'spacer3']
173534:  i=3
Traceback (most recent call last):
  File "C:\Users\caver\Documents\GitHub\radiolog\radiolog.py", line 5604, in tabContextMenu
    self.deleteTeamTab(niceTeamName)
  File "C:\Users\caver\Documents\GitHub\radiolog\radiolog.py", line 6117, in deleteTeamTab
    self.deleteTeamTab(self.extTeamNameList[n+1],True)
  File "C:\Users\caver\Documents\GitHub\radiolog\radiolog.py", line 6081, in deleteTeamTab
    del self.ui.tabList[i]
        ~~~~~~~~~~~~~~~^^^
IndexError: list assignment index out of range
```

But, during various hide and unhide testing, it was apparent that back-to-back spacers would visually show up in the tab bar - probably during unhide - and hopefully the other errors are due to the back-to-back spacers.  Maybe find a way to prevent back-to-back spacers on unhide...?

**Comment by caver456** [comment]
These cases are obscure enough that I wrote a user-action-summarizer that reads a radiolog transcript and shows a short summary of user actions leading up to / including tracebacks.

Until/unless a bigger overhaul of the various list-syncing is done, to keep the list of displayed tabs as intended, this game of whack-a-mole might continue...

Whenever a traceback shows up during hiding-and-unhiding tests, capture the summary and try to get a minimal reproducible test case.  Then look at the corresponding full transcript to sleuth it out.

Next up - in case there's need to track these various errors, let's call the one-time tracebacks by number, and the repeating tracebacks by letter, so this will be 1A:

```
203446:newEntry 'Team 101', 'STARTING ASSIGNMENT', None, 'Working', 1727062484.357744, None, None, None]
203454:newEntry '1p42', 'STARTING ASSIGNMENT', None, 'Working', 1727062490.56877, None, None, None]
203502:newEntry '4p13', 'STARTING ASSIGNMENT', None, 'Working', 1727062498.9025872, None, None, None]
203513:deleting team tab: teamName=4p13  extTeamName=z_04p13  niceTeamName=4p13
203513:deleting team tab: teamName=spacer2  extTeamName=spacer2  niceTeamName=Spacer 2
203518:newEntry 3', '[RADIOLOG: operator is unhiding hidden team tab for "4p13"]', '', 'Working', 1727062518.2315185, '', '', '']
203521:deleting team tab: teamName=1p42  extTeamName=z_01p42  niceTeamName=1p42
203521:deleting team tab: teamName=spacer1  extTeamName=spacer1  niceTeamName=Spacer 1
203524:newEntry 2', '[RADIOLOG: operator is unhiding hidden team tab for "1p42"]', '', 'Working', 1727062524.60886, '', '', '']
203527:deleting team tab: teamName=Team 101  extTeamName=z_Team00101  niceTeamName=Team 101
203527:deleting team tab: teamName=spacer3  extTeamName=spacer3  niceTeamName=Spacer 3
203530:deleting team tab: teamName=1p42  extTeamName=z_01p42  niceTeamName=1p42
203530:deleting team tab: teamName=spacer1  extTeamName=spacer1  niceTeamName=Spacer 1
203533:newEntry 'Team 101', 'STARTING ASSIGNMENT', None, 'Working', 1727062532.1516137, None, None, None]
Traceback (most recent call last):
  File "C:\Users\caver\Documents\GitHub\radiolog\radiolog.py", line 7698, in keyPressEvent
    self.accept()
  File "C:\Users\caver\Documents\GitHub\radiolog\radiolog.py", line 7851, in accept
    self.parent.newEntry(val,self.amendFlag)
  File "C:\Users\caver\Documents\GitHub\radiolog\radiolog.py", line 5002, in newEntry
    self.newEntryProcessTeam(niceTeamName,status,values[1],values[3],amend,unhiding=unhiding)
  File "C:\Users\caver\Documents\GitHub\radiolog\radiolog.py", line 5017, in newEntryProcessTeam
    self.newTeam(niceTeamName,unhiding=unhiding)
  File "C:\Users\caver\Documents\GitHub\radiolog\radiolog.py", line 5246, in newTeam
    self.addTab(extTeamName)
  File "C:\Users\caver\Documents\GitHub\radiolog\radiolog.py", line 5375, in addTab
    self.ui.tabList[i].setStyleSheet('font-size:'+str(self.menuFontSize)+'pt')
    ~~~~~~~~~~~~~~~^^^
IndexError: list index out of range
Traceback (most recent call last):
  File "C:\Users\caver\Documents\GitHub\radiolog\radiolog.py", line 4064, in updateTeamTimers
    f=self.ui.tabWidget.tabBar().tabButton(i,QTabBar.LeftSide).font()
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'font'
...(repeating)
```

**Comment by caver456** [comment]
1A: this sequence looks like the problem - extTeamNameList after sort still respects the fact that 1p42 is hidden, but, it adds an extra spacer anyway:

```
062322:  extTeamNameList after delete 2: ['spacerLeft', 'z_04p13', 'spacer2']
062333:non-team-hotkey "1" pressed; calling openNewEntry
062333:openNewEntry called:key=1 callsign=None formattedLocString=None fleet=None dev=None origLocString=None amendFlag=False amendRow=None isMostRecentForCallsign=False
062333:newEntryWidget __init__ called: formattedLocString=None fleet=None dev=None origLocString=None amendFlag=False amendRow=None isMostRecentForCallsign=False
062333:ERROR in call to getCallsign: fleetOrId is not a string.
062333:newEntryWidget.addTab called: labelText=0623  widget=<__main__.newEntryWidget object at 0x000001D48410AF90>
062333:inserting tab
062335:Accepted
062335:newEntry called with these values:
062335:['0623', 'FROM', 'Team 101', 'STARTING ASSIGNMENT', None, 'Working', 1727097813.8226285, None, None, None]
062335:new team: newTeamName=Team 101 extTeamName=z_Team00101 niceTeamName=Team 101 shortNiceTeamName=101
062335:extTeamNameList before sort:['spacerLeft', 'z_04p13', 'spacer2', 'z_Team00101']
062335:grouped tabs:{'NCSOcmd': [], 'NCSO': ['z_01p42'], 'NCPD': [], 'TPD': [], 'GVPD': ['z_04p13'], 'CHP': [], 'Numbers': ['z_Team00101'], 'other': []}
062335:extTeamNameList after sort:['spacerLeft', 'spacer1', 'z_04p13', 'spacer2', 'z_Team00101', 'spacer3']
```

**Comment by caver456** [comment]
1A fixed by modifying rebuildGroupedTabDict to not add a spacer if all group members are hidden.

After that, then adding a typed-callsign entry from 1p42, 1p42 shows up with no spacer after it and A begins, without any one-time traceback before it.

**Comment by caver456** [comment]
related to B: with a more minimal test case (add Team 101; add 1p42; hide 1p42; type 1p42) no spacer appears between 1p42 and 101, and right-clicking 101 does nothing, but there are no tracebacks.  Guessing this will be solved by the same fix as B: during newEntry (could alternately be done in accept or newEntryProcessTeam - any of these would work), if extTeamName is in hiddenTeamTabsList, force unhiding to True.

**Comment by caver456** [comment]
Confirmed that the problems in the previous comment are fixed as above.

Also confirmed that B is fixed.

Also confirmed that the case in the initial issue description is fixed.

Committing now, but should also check the full script for A1 (and B) using only typed-callsigns, and again using fleetsync.

**Comment by caver456** [comment]
All QA is done for typed-callsigns-only and for fleetsync-only.  Looks like it works!  Time to cleanup and merge.

**Comment by caver456** [comment]
After closing this issue and merging its fix, I did some more QA of the recent MRA mock which led to #741.

Used the summarizer script to extract the actions done on the effort to reproduce those issues, and confirmed that these same actions don't cause any tracebacks or unexpected displays.  Also confirmed that this script DID cause the same errors as reported in #741 when run on 3.11.5, the version used for the MRA.

These errors had already been fixed by the time of 3.11.2, so, this additional QA is just ensuring that the fixes on this issue didn't screw anything up.

Here's the summary of the MRA reproduction test 2024_08_20_202059:

```
202136:newEntry 'RADIO OPERATOR LOGGED IN: Grundy, Tom  SAR35', '', '', 1724210496.9899404, '', '', '']
202816:newEntry 'Team 101', 'STARTING ASSIGNMENT', None, 'Working', 1724210895.227438, None, None, None]
202818:newEntry 'Team 102', 'STARTING ASSIGNMENT', None, 'Working', 1724210897.7042427, None, None, None]
202821:newEntry 'Team 103', 'STARTING ASSIGNMENT', None, 'Working', 1724210899.3690376, None, None, None]
202831:newEntry 'Marin', 'STARTING ASSIGNMENT', None, 'Working', 1724210907.5037, None, None, None]
202845:newEntry 'Medical1', 'STARTING ASSIGNMENT', None, 'Working', 1724210917.0348155, None, None, None]
202851:newEntry 'Medical2', 'STARTING ASSIGNMENT', None, 'Working', 1724210926.0117998, None, None, None]
202900:newEntry 'Medical101', 'STARTING ASSIGNMENT', None, 'Working', 1724210934.9373097, None, None, None]
202919:newEntry 'Rescue1', 'STARTING ASSIGNMENT', None, 'Working', 1724210946.1243918, None, None, None]
202926:newEntry 'Rescue101', 'STARTING ASSIGNMENT', None, 'Working', 1724210960.5898893, None, None, None]
203736:deleting team tab: teamName=Team 101  extTeamName=z_Team00101  niceTeamName=Team 101
203742:deleting team tab: teamName=Team 102  extTeamName=z_Team00102  niceTeamName=Team 102
203748:deleting team tab: teamName=Team 103  extTeamName=z_Team00103  niceTeamName=Team 103
203748:deleting team tab: teamName=spacer1  extTeamName=spacer1  niceTeamName=Spacer 1
203805:deleting team tab: teamName=Marin  extTeamName=z_Marin  niceTeamName=Marin
203812:deleting team tab: teamName=Rescue 101  extTeamName=z_Rescue00101  niceTeamName=Rescue 101
203847:newEntry 'Rescue 101', 'COMPLETED ASSIGNMENT', None, '', 1724211507.0782287, None, None, None]
Traceback (most recent call last):
  File "C:\Users\caver\Documents\GitHub\radiolog\radiolog.py", line 7559, in accept
    self.parent.newEntry(self.getValues(),self.amendFlag)
  File "C:\Users\caver\Documents\GitHub\radiolog\radiolog.py", line 4806, in newEntry
    self.newEntryProcessTeam(niceTeamName,status,values[1],values[3],amend)
  File "C:\Users\caver\Documents\GitHub\radiolog\radiolog.py", line 4821, in newEntryProcessTeam
    self.newTeam(niceTeamName)
  File "C:\Users\caver\Documents\GitHub\radiolog\radiolog.py", line 5029, in newTeam
    self.addTab(extTeamName)
  File "C:\Users\caver\Documents\GitHub\radiolog\radiolog.py", line 5154, in addTab
    self.ui.tabList[i].setStyleSheet('font-size:'+str(self.menuFontSize)+'pt')
    ~~~~~~~~~~~~~~~^^^
IndexError: list index out of range
Traceback (most recent call last):
  File "C:\Users\caver\Documents\GitHub\radiolog\radiolog.py", line 3873, in updateTeamTimers
    button.setStyleSheet(buildObjSS(styleObjectName,statusStyleDict[status]))
    ^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'setStyleSheet'
Traceback (most recent call last):
  File "C:\Users\caver\Documents\GitHub\radiolog\radiolog.py", line 3873, in updateTeamTimers
    button.setStyleSheet(buildObjSS(styleObjectName,statusStyleDict[status]))
    ^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'setStyleSheet'
...
```
